controller/controlplane/controller_reconciler_utils.go

```diff
@@ -764,3 +797,70 @@ func (r *Reconciler) ensureValidatingWebhookConfiguration(
 
 	return op.Created, r.Client.Create(ctx, generatedWebhookConfiguration)
 }
+
+func (r *Reconciler) validateReferenceGrants(
+	ctx context.Context,
+	cp *operatorv1beta1.ControlPlane,
+) error {
+	if cp.Spec.WatchNamespaces == nil {
+		return nil
+	}
```

Comment:
This technically shouldn't happen due to the default on the CRD, right? I think this will be more future-proof if we return an error here, as this is something we do not expect.
```suggestion
	if cp.Spec.WatchNamespaces == nil {
		return errors.New("spec.watchNamespaces cannot be empty")
	}
```